src/viam/module/module.py

```diff
@@ -57,13 +69,40 @@ def from_args(cls) -> Self:
         Returns:
             Module: a new Module instance
         """
+        # todo(review): change this to use parse_module_args()? makes life simpler below
         args = sys.argv
         if len(args) < 2:
             raise Exception("Need socket path as command line argument")
         address = args[1]
         log_level = logging.DEBUG if (len(args) == 3 and "=debug" in args[2].lower()) else logging.INFO
         return cls(address, log_level=log_level)
 
+    @classmethod
+    async def run_with_models(cls, *models: ResourceBase):
+        """
+        Module entrypoint that takes a list of ResourceBase subclasses.
+        In most cases you'll want to use run_from_registry instead (see below).
+        """
+        args = parse_module_args()
+        module = cls(args.socket_path, log_level=args.log_level)
+        for model in models:
+            # todo(review): the pyright ignore below is for ResourceBase.MODEL. Is this not a required field? Should I not rely on it?
+            module.add_model_from_registry(model.SUBTYPE, model.MODEL)  # pyright: ignore [reportAttributeAccessIssue]
+        await module.start()
+
+    @classmethod
+    async def run_from_registry(cls):
+        """
+        Module entrypoint that automatically includes all the resources you've created in your program.
+        For usage, see the bottom of examples/easy_resource/main.py.
```

Comment:
Would be good to have usage instructions here, rather than forcing someone to go to the example